skrub/_interpolation_joiner.py

```diff
@@ -287,43 +291,46 @@ def transform(self, X):
             main_table, self._main_key, "'X' (the main table)"
         )
         key_values = self.vectorizer_.transform(
-            main_table[self._main_key].set_axis(self._aux_key, axis="columns")
+            sbd.set_column_names(
+                s.select(main_table, s.cols(*self._main_key)), self._aux_key
+            )
         )
         prediction_results = joblib.Parallel(self.n_jobs)(
             joblib.delayed(_predict)(
+                main_table,
                 key_values,
                 assignment["columns"],
                 assignment["estimator"],
                 propagate_exceptions=(self.on_estimator_failure == "raise"),
             )
             for assignment in self.estimators_
         )
-        prediction_results = self._check_prediction_results(prediction_results)
+        prediction_results = self._check_prediction_results(
+            main_table, prediction_results
+        )
         predictions = [res["predictions"] for res in prediction_results]
         predictions = [
             _join_utils.add_column_name_suffix(df, self.suffix) for df in predictions
         ]
-        for part in predictions:
-            part.index = main_table.index
-        return pd.concat([main_table] + predictions, axis=1)
+        return sbd.concat_horizontal(main_table, *predictions)
```

Comment:
now we have a different index than in the input for pandas dataframes. either here or in the `concat_horizontal` it might be good to preserve the index of the first dataframe, WDYT?

Reply:
I agree, let's do this in the `concat_horizontal` so that we have a consistent behavior if we use it elsewhere